02-express-tutorial/app.js

```diff
@@ -1 +1,67 @@
-console.log('Express Tutorial')
+const express = require('express')
+const app = express()
+const { products } = require('./data')
+
+app.use(express.static('./public'))
+
+app.get('/api/v1/products', (req, res) => {
+	const newProducts = products.map((item) => {
+		const { id, name, price } = item
+		return { id, name, price }
+	})
+	res.json(newProducts)
+})
+
+app.get('/api/v1/products/:productID', (req, res) => {
+	console.log(res)
+	const { productID } = req.params
+
+	const productByID = products.find(
+		(product) => product.id === parseInt(productID)
+	)
+	if (!productByID) {
+		return res.status(404).json({ message: 'Product not found!' })
+	} else {
+		res.json(productByID)
+	}
+})
+
+app.get('/api/v1/query', (req, res) => {
+	const { search, limit, price } = req.query
+	let sortedProducts = [...products]
+
+	if (price) {
+		sortedProducts = sortedProducts.filter((product) => {
+			const productPrice = product.price
+			console.log(productPrice)
+			return (
+				productPrice >= parseFloat(price) && productPrice <= parseFloat(price)
```

Comment:
This condition will match only products with prices equal to the given one. If the idea is to match products that cost less, you need to check for `productPrice <= parseFloat(price)` instead.